src/llmq/ehf_signals.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2023 The Dash Core developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_LLMQ_EHF_SIGNALS_H
+#define BITCOIN_LLMQ_EHF_SIGNALS_H
+
+#include <llmq/signing.h>
+
+#include <set>
+
+class CBlockIndex;
+class CChainState;
+class CConnman;
+class CMNHFManager;
+class CSporkManager;
+class CTxMemPool;
+
+namespace llmq
+{
+class CQuorumManager;
+class CSigSharesManager;
+class CSigningManager;
+
+class CEHFSignalsHandler : public CRecoveredSigsListener
+{
+private:
+    CChainState& chainstate;
+    CConnman& connman;
+    CSigningManager& sigman;
+    CSigSharesManager& shareman;
+    CSporkManager& sporkman;
+    CQuorumManager& qman;
+    CTxMemPool& mempool;
+    CMNHFManager& mnhfManager;
```

Comment:
These can be const (and change constructor)
```suggestion
    const CSporkManager& sporkman;
    const CQuorumManager& qman;
    CTxMemPool& mempool;
    const CMNHFManager& mnhfManager;
```